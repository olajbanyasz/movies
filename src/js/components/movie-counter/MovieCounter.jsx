import React from 'react';
import './moviecounter.style.less';

const getCounterText = (result) => `${result} ${result === 1 ? 'movie' : 'movies'} found`;

function MovieCounter(props) {
  return (
    <div className='movie-counter'>
      {getCounterText(props.resultcounter)}
    </div>
  );
};

export default MovieCounter;