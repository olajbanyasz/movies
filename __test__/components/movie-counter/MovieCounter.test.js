import React from 'react';
import MovieCounter from '../../../src/js/components/movie-counter/MovieCounter';
import { mount } from 'enzyme';

describe('MovieCounter', () => {
  it('should rendered correctly', () => {
    const component = mount(<MovieCounter resultcounter={1} />);
    expect(component.props().resultcounter).toBe(1);
    expect(component).toMatchSnapshot();
    expect(component.text()).toBe('1 movie found');
  });
});