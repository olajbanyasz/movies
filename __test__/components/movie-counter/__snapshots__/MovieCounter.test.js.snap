// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCounter should rendered correctly 1`] = `
ReactWrapper {
  Symbol(enzyme.__unrendered__): <MovieCounter
    resultcounter={1}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "resultcounter": 1,
    },
    "ref": null,
    "rendered": Object {
      "instance": <div
        class="movie-counter"
      >
        1 movie found
      </div>,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "1 movie found",
        "className": "movie-counter",
      },
      "ref": null,
      "rendered": Array [
        "1 movie found",
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "resultcounter": 1,
      },
      "ref": null,
      "rendered": Object {
        "instance": <div
          class="movie-counter"
        >
          1 movie found
        </div>,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "1 movie found",
          "className": "movie-counter",
        },
        "ref": null,
        "rendered": Array [
          "1 movie found",
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
